admin-ui/src/component/my-units/create-reservation/CreateReservationModal.tsx

```diff
@@ -0,0 +1,290 @@
+`;
+
+const DialogContent = ({
+  onClose,
+  reservationUnit,
+  start,
+}: {
+  onClose: () => void;
+  reservationUnit: ReservationUnitType;
+  start: Date;
+}) => {
+  const { t } = useTranslation();
+  const form = useForm<ReservationFormType>({
+    resolver: joiResolver(
+      reservationSchema(reservationUnit.reservationStartInterval)
+    ),
+    shouldFocusError: true,
+    defaultValues: {
+      date: valueForDateInput(start.toISOString()),
+      startTime: formatDate(start.toISOString(), "HH:mm") as string,
+      bufferTimeBefore: false,
+      bufferTimeAfter: false,
+    },
+  });
+
+  const {
+    formState: { errors },
+  } = form;
+
+  const { notifyError, notifySuccess } = useNotification();
+
+  const type = form.watch("type");
+
+  const [create] = useMutation<
+    { createStaffReservation: ReservationStaffCreateMutationPayload },
+    { input: ReservationStaffCreateMutationInput }
+  >(CREATE_STAFF_RESERVATION);
+
+  const createStaffReservation = (input: ReservationStaffCreateMutationInput) =>
+    create({ variables: { input } });
+
+  const renamePkFields = ["ageGroup", "homeCity", "purpose"];
+
+  const onSubmit = async () => {
+    try {
+      const metadataSetFields = (
+        (reservationUnit.metadataSet?.supportedFields || []) as string[]
+      ).map(camelCase);
+
+      const metadataSetValues = pick(form.getValues(), metadataSetFields);
+
+      const flattenedMetadataSetValues = zipObject(
+        Object.keys(metadataSetValues).map((k) =>
+          renamePkFields.includes(k) ? `${k}Pk` : k
+        ),
+        Object.values(metadataSetValues).map((v) => get(v, "value") || v)
+      );
+
+      const input = {
+        reservationUnitPks: [reservationUnit.pk as number],
+        type: String(form.getValues("type")),
+        begin: dateTime(form.getValues("date"), form.getValues("startTime")),
+        end: dateTime(
+          form.getValues("date"),
+          form.getValues("endTime") as string
+        ),
+        bufferTimeBefore: form.getValues("bufferTimeBefore")
+          ? String(reservationUnit.bufferTimeBefore)
+          : undefined,
+        bufferTimeAfter: form.getValues("bufferTimeAfter")
+          ? String(reservationUnit.bufferTimeAfter)
+          : undefined,
+        workingMemo: form.getValues("workingMemo"),
+        ...flattenedMetadataSetValues,
+      } as ReservationStaffCreateMutationInput;
+
+      await createStaffReservation(input);
+      notifySuccess(
+        t("ReservationDialog.saveSuccess", {
+          reservationUnit: reservationUnit.nameFi,
+        })
+      );
+      onClose();
+    } catch (e) {
+      notifyError(
+        t("ReservationDialog.saveFailed", { error: get(e, "message") })
+      );
+    }
+  };
+
+  return (
+    <>
+      <Dialog.Content>
+        <form onSubmit={form.handleSubmit(onSubmit)}>
+          <VerticalFlex style={{ marginTop: "var(--spacing-m)" }}>
+            <CommonFields>
+              <Controller
```

Comment:
Näisäkö ei "...register()" toimi? Ymmärrän kyl että custom komponenttien kanssa tämä on aina vähän ongelma

Reply:
jep, ongelmaa oli ja siks näin